src/bin/glarea/main.rs

```diff
@@ -0,0 +1,145 @@
+//! # GLArea Sample
+//!
+//! This sample demonstrates how to use the GLArea widget with glium to
+//! manage the rendering.
+
+extern crate gio;
+extern crate glib;
+extern crate glium;
+extern crate gtk;
+extern crate shared_library;
+
+use gio::prelude::*;
+use glium::Surface;
+use gtk::prelude::*;
+use gtk::WidgetExt;
+
+mod glium_backend;
+
+#[derive(Copy, Clone)]
+struct Vertex {
+    position: [f32; 2],
```

Comment:
It's nitpicking but why not creating a `Position` type and a `Color` type?

Reply:
Not done, because I think it's overkill. Also, I'd have to check if it played nicely with the `implement_vertex!` macro. Also, it's important that the layout of these types match in the shaders, and it's more cumbersome to verify that if they're newtypes.

Reply:
Ok!